docs/opentracing.rst

```diff
@@ -0,0 +1,179 @@
+===========
+Opentracing
+===========
+
+Background
+----------
+
+Opentracing is semi-standard being addopted by a number of distributed tracing
+platforms. It is a standardised api for facilitating vendor agnostic tracing
+instrumentation. That is, we can use the opentracing api and select one of a
+number of tracer implementations to do the heavy lifting in the background.
+Our current selected implementation is Jaeger.
+
+Opentracing concepts can be found at
+https://opentracing.io/docs/overview/what-is-tracing/
+
+Python specific tracing concepts are at https://opentracing.io/guides/python/.
```

Comment:
I'd like us to keep developer documentation separate from user (documentation). At the moment, most of this is stuff that is useful to know to someone who is running synapse, but not writing any code, but I think this paragraph is probably a bit unhelpful to a user rather than a developer.

Developer docs could end up in the same file, but it would need to be in a separate section, rather than mixed together. (Another option would be to put it in a docstring in `opentracing.py` or something.)

Reply:
Is there an upper limit to the amount of documentation we put in a docstring? I haven't seen many long doc strings in the code base. Including the entire `Using opentracing in synapse` section in a docstring seems like a lot.

Reply:
There's no limit at all; people write whole books in there, especially for file-level docstrings.

Reply:
The unsung sagas. I've just moved it to the docstring verbatim.